cmd/podman/cp.go

```diff
@@ -190,53 +178,45 @@ func copyBetweenHostAndContainer(runtime *libpod.Runtime, src string, dest strin
 			}
 			srcPath = path
 		} else if isBindMount, mount := isBindMountDestName(srcPath, ctr); isBindMount {
-			path, err := pathWithBindMountSource(mount, srcPath)
-			if err != nil {
-				return errors.Wrapf(err, "error getting source path from bind moutn %s", mount.Destination)
-			}
-			srcPath = path
+			srcPath = pathWithBindMountSource(mount, srcPath)
 		} else if filepath.IsAbs(srcPath) {
-			cleanedPath, err := securejoin.SecureJoin(mountPoint, srcPath)
-			if err != nil {
-				return err
-			}
-			srcPath = cleanedPath
-		} else {
-			cleanedPath, err := securejoin.SecureJoin(mountPoint, filepath.Join(ctr.WorkingDir(), srcPath))
-			if err != nil {
-				return err
-			}
-			srcPath = cleanedPath
+			srcPath = filepath.Join(mountPoint, srcPath)
 		}
 	}
-	glob, err = filepath.Glob(srcPath)
-	if err != nil {
-		return errors.Wrapf(err, "invalid glob %q", srcPath)
-	}
-	if len(glob) == 0 {
-		glob = append(glob, srcPath)
-	}
-	if !filepath.IsAbs(destPath) {
+
+	if !(filepath.IsAbs(destPath) || destPath == "-") {
 		dir, err := os.Getwd()
 		if err != nil {
 			return errors.Wrapf(err, "err getting current working directory")
 		}
 		destPath = filepath.Join(dir, destPath)
 	}
 
-	var lastError error
-	for _, src := range glob {
-		if src == "-" {
-			src = os.Stdin.Name()
-			extract = true
-		}
-		err := copy(src, destPath, dest, idMappingOpts, &destOwner, extract, isFromHostToCtr)
-		if lastError != nil {
-			logrus.Error(lastError)
+	if srcPath, err = cleanPath(srcPath); err != nil {
+		return err
+	}
+
+	if destPath, err = cleanPath(destPath); err != nil {
+		return err
+	}
```

Comment:
This can redirect the paths from inside the container to outside. How is the use of such paths prevented?

Reply:
… and why aren’t there any tests that fail if this exploit succeeds? Maintaining the security of the chroot is a critical feature of the `cp` functionality.